plugins/terms/plugin_tests/termsSpec.js

```diff
@@ -65,12 +67,20 @@ describe('Navigate to a non-collection folder and item', function () {
         waitsFor(function () {
             return $('.g-item-count-container:visible').length === 1;
         });
+        girderTest.waitForLoad();
         runs(function () {
             expect($('.g-hierarchy-breadcrumb-bar>.breadcrumb>.active').text()).toBe('Public');
             var folderId = window.location.hash.split('/')[3];
             expect(folderId).toMatch(/[0-9a-f]{24}/);
             window.location.assign('#folder/' + folderId);
         });
+        // after setting a window location, waitForLoad is insufficient, as the
+        // page hasn't yet started making requests and it looks similar to
+        // before the location change.  Wait for the user header to be hidden,
+        // and then wait for load.
+        waitsFor(function () {
```

Comment:
I suppose it's theoretically possible to have another race condition here, whereby the navigation happens so fast that the `waitsFor` never gets to poll it during the incomplete state? However, I admit that under the current test conditions, that's highly unlikely.

Reply:
It can't happen, since we were at `#user/id/folder/id` and go to `#folder/id`, and the .g-user-header element will no longer be present after that (it is the element that lists the user when `#user` is part of the route).